### PR TITLE
Add `starknet_addDeclareTransaction` JSON-RPC endpoint

### PR DESCRIPTION
**Pull request by kkovaacs**
This JSON-RPC endpoint is a proxy that forwards data received to the Starknet sequencer, just like other write API methods we have.

This PR also adds some missing fields to the `sequencer::reply::transaction::Transaction` type that are required so that we can parse the new DECLARE transaction JSON.

### CODE REVIEW

File: crates/pathfinder/src/core.rs [hunk 1]
```diff
@@ -10,6 +10,10 @@ use web3::types::{H128, H160, H256};
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Deserialize, Serialize, PartialOrd, Ord)]
 pub struct ContractAddress(pub StarkHash);
 
+/// The hash of a StarkNet class.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Deserialize, Serialize, PartialOrd, Ord)]
+pub struct ClassHash(pub StarkHash);
+
```
Comment (Mirko-von-Leipzig) [inline crates/pathfinder/src/core.rs:16]: I think this is the same as `ContractHash`? We should consider renaming things to be align with the new class concepts.

**Review by Mirko-von-Leipzig — APPROVED** [review]

**Review by CHr15F0x — APPROVED** [review]
Would be nice to have an example added too

File: crates/pathfinder/src/core.rs (hunk 1 above)
Comment (koivunej) [reply crates/pathfinder/src/core.rs:16]: > constructor arguments

Where do you see this? Entry point is not an argument.

File: crates/pathfinder/src/core.rs [hunk 2]
```diff
@@ -157,6 +161,10 @@ pub struct Fee(pub H128);
 #[derive(Debug, Copy, Clone, PartialEq, Deserialize, Serialize)]
 pub struct GasPrice(pub u128);
 
+// Starknet transaction nonce.
+#[derive(Debug, Copy, Clone, PartialEq, Deserialize, Serialize)]
+pub struct Nonce(pub StarkHash);
```
[the remark this answers is not in the document]
Comment (koivunej) [reply crates/pathfinder/src/core.rs:166]: Yeah I think this would be a good idea.

[the remark this answers is not in the document]
Comment (kkovaacs) [reply crates/pathfinder/src/core.rs:166]: Indeed, it's a good idea.

File: crates/pathfinder/src/core.rs (hunk 1 above)
Comment (koivunej) [reply crates/pathfinder/src/core.rs:16]: This PR is already 10 files so I'd go with type alias or something, with a todo to rename all...?

**Review by koivunej — APPROVED** [review]

**Pull request merged**